c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2020,7 +2023,8 @@ class DeviceCachingAllocator {
           return a.address < b.address;
         });
 
-    record_trace(TraceEntry::SNAPSHOT, 0, total_active, nullptr, 0, nullptr);
+    record_trace(
+        TraceEntry::SNAPSHOT, 0, total_active, nullptr, 0, nullptr, nullptr);
```

Comment:
Should the `nullptr` for the BlockPool argument of `record_trace` here be: `head_block->pool` instead, so that `record_trace` derives the correct id (instead of just {0,0}) for this snapshot trace?

Reply:
Good point! We can't use `head_block` (it's just an iterator over `all_blocks`) and, in fact, the current interface of `record_trace` doesn't make it easy since it expects a `BlockPool*`, and each private mempool has two of those (large and small). However, at the end of the day, all `record_trace` needs is a `MempoolId_t`, and this we can provide even for snapshot, so I changed the interface of `record_trace` and managed to get this done!